### PR TITLE
2363 refactor the dialog component to use the new scope architecture

### LINKED ISSUE #2363: Refactor the Dialog component to use the new scope architecture
## Scope & Context

We are doing a global refactoring to introduce components scopes.

## Technical inputs

- Refactor DialogProvider to DialogManager
- Create `DialogManagerScope`
- Create a useDialogManager() hook to expose dialogInternal and setDialogInternal
- Create a DialogEffect component to apply this effect:
```
useEffect(() => {
    if (dialogInternal.queue.length === 0) {
      return;
    }

    setHotkeyScopeAndMemorizePreviousScope(DialogHotkeyScope.Dialog);
  }, [dialogInternal.queue, setHotkeyScopeAndMemorizePreviousScope]);
```

### PR DESCRIPTION
**Pull request by bosiraphael**
Closes #2363

### CODE REVIEW

File: front/src/modules/ui/feedback/dialog/hooks/internal/useDialogManagerScopedStates.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { useAvailableScopeIdOrThrow } from '@/ui/utilities/recoil-scope/scopes-internal/hooks/useAvailableScopeId';
+
+import { DialogManagerScopeInternalContext } from '../../scopes/scope-internal-context/DialogManagerScopeInternalContext';
+import { dialogInternalScopedState } from '../../states/dialogInternalScopedState';
+
+type useDialogInternalProps = {
+  dialogManagerScopeId?: string;
+};
+
+export const useDialogInternal = (props?: useDialogInternalProps) => {
```
Comment (charlesBochet) [inline front/src/modules/ui/feedback/dialog/hooks/internal/useDialogManagerScopedStates.ts:11]: naming?

File: front/src/modules/ui/feedback/dialog/components/DialogManager.tsx [hunk 2]
```diff
@@ -1,30 +1,18 @@
 import { useEffect } from 'react';
```
Comment (charlesBochet) [inline front/src/modules/ui/feedback/dialog/components/DialogManager.tsx:1]: I feel the whole folder should be renamed to DialogManager and we should only have useDialogManager (not useDialog)

File: front/src/modules/ui/feedback/dialog/types/DialogOptions.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { DialogProps } from '../components/Dialog';
+
+export type DialogOptions = DialogProps & {
+  id: string;
```
Comment (charlesBochet) [inline front/src/modules/ui/feedback/dialog/types/DialogOptions.ts:4]: what is is?

Comment (bosiraphael) [reply front/src/modules/ui/feedback/dialog/types/DialogOptions.ts]: It is used to type the elements in the queue

**Review by charlesBochet — APPROVED** [review]
LGTM !

**Pull request merged**